source/common/grpc/typed_async_client.h

```diff
@@ -74,6 +83,42 @@ template <typename Response> class AsyncRequestCallbacks : public RawAsyncReques
   }
 };
 
+/**
+ * A versioned gRPC client.
+ */
+class VersionedClient {
```

Comment:
Is this really a client? Trying to get at whether this is the right interface, vs. maybe `VersionedMethod`?